src/hooks/useProfile.tsx

```diff
@@ -0,0 +1,71 @@
+import axios from 'axios';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import create from 'zustand';
+
+import { API_URL } from '../config/config';
+import { immer } from '../utils/zustand';
+
+export type UserProfile = {
+  bio: string | undefined;
+  userName: string | undefined;
+  profilePicture: string | undefined;
+  deliveryTime: number | undefined;
+  demos: string[] | undefined;
+  setBio: (bio: string) => void;
+  setUsername: (username: string) => void;
+  setProfilePicutre: (profilePicture: string) => void;
+  setDeliveryTime: (deliveryTime: number) => void;
+  setDemos: (demos: string[]) => void;
+  verifyUser: (tweetUrl: string, address: string) => Promise<void>;
+};
+
+export const useProfile = create<UserProfile>(
```

Comment:
I have no idea if this is how I'm supposed to use immer.

Reply:
this works for now!! we can refactor as needed